File: targets/python-web/vulns/vuln10_ssrf.py
import requests
from flask import Blueprint, request
bp = Blueprint("vuln10", __name__)

@bp.get("/vuln10/fetch")
def fetch():
    url = request.args.get("url", "")
    # VULN:VULN-10:CWE-918:taint SSRF — server fetches an arbitrary user-supplied URL
    return requests.get(url, timeout=4).text

File: targets/jsts-web/src/vulns/vuln10_ssrf.ts
import { Router } from "express";
import axios from "axios";
export const router = Router();
router.get("/vuln10/fetch", async (req, res) => {
  const url = String(req.query.url || "");
  // VULN:VULN-10:CWE-918:taint SSRF — server fetches an arbitrary user-supplied URL
  try { const r = await axios.get(url, { timeout: 4000 }); res.send(typeof r.data === "string" ? r.data : JSON.stringify(r.data)); }
  catch (e: any) { res.send("error: " + e.message); }
});
